### ISSUE TITLE
[Accessibility][Web] Main element should be `<main>` instead of `<div id="main">`

### ISSUE THREAD

**Issue by bandogora** [DioxusLabs/dioxus#5334]
**Problem**

The main element created by Dioxus is `<div id="main" data-dioxus-id="0">`, which gives the wrong [ARIA role type](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Reference/Roles#aria_role_types): "generic".

This could be fixed by simply adding the correct role, but [according to MDN](https://developer.mozilla.org/en-US/docs/Web/HTML/Reference/Elements/main#landmark) we should "Prefer using the `<main>` element over declaring `role="main"`, unless there are legacy browser support concerns."

As far as I can tell there is no way to manually correct this as the div is created by Dioxus, not the user.

**Steps To Reproduce**

- Run `dx serve --web`
- Open browser
- Check for Accessibility issues in dev tools

**Expected behavior**

Main element should be `<main>` or at least have the ARIA role type "main".

**Screenshots**

<img width="1637" height="483" alt="Accessibility view" src="https://github.com/user-attachments/assets/bcca0942-f97f-4a69-9712-a171721b3202" />

<img width="585" height="313" alt="Inspector view" src="https://github.com/user-attachments/assets/2ce06490-b074-495f-a635-adde7c913695" />

**Environment:**

- Dioxus version:  0.7.3
- Rust version:    rustc 1.93.1 (01f6ddf75 2026-02-11), stable
- OS info:         stable-x86_64-unknown-linux-gnu
- App platform:    Web
- Web browser:     Firefox

**Questionnaire**

I don't have time to fix this right now, but maybe later.

**Comment by ealmloff** [comment]
Dioxus renders all content inside that div. Changing it to `main` would prevent adding elements outside of main like navigation and header content. From the same page:
```html
<body>
  <a href="#main-content">Skip to main content</a>

  <!-- navigation and header content -->

  <div id="main-content" role="main">
    <!-- main page content -->
  </div>
</body>
```

Is there any issues with rendering the `main {}` element in rsx inside of the existing roleless div?

**Comment by bandogora** [comment]
Ah, you're right. I missed [the part](https://developer.mozilla.org/en-US/docs/Web/HTML/Reference/Elements/main#usage_notes) about not having navigation in `main`.

> Is there any issues with rendering the `main {}` element in rsx inside of the existing roleless div?

No, other than the alleged accessibility issue from Firefox. I suppose this is more of a potential best practice than an issue.

I haven't looked too much at Dioxus' internals, but I see two possible solutions:

1. Don't create the `div` and apply `data-dioxus-id` to the `body`
2. Apply `role="none"` to the top level div
    - I don't think this would affect the descendants in this case, but I'm not positive.

**Comment by ealmloff** [comment]
Dioxus will attach itself to any element with `id="main"`. You can provide a custom index.html with a different id if you want to customize this for your application.

I don't think divs have any meaningful default role applying role `"none"` would change. Closing this for now as not planned since moving everything into `main` would hurt accessibility.

If you can provide examples where the current structure hurts accessibility in applications that implement accessibility roles (they have their own `main {}`), we could consider changing this to directly render into the body, but that would complicate streaming ssr rendering.
